### PR TITLE
Update README.

### PR DESCRIPTION
**Pull request by dan-zheng**
- Minor reorganization, formatting changes, stylistic edits.
- Add links to referenced files.

Happened to update the README while looking into build updates for SwiftFusion!

### CODE REVIEW

File: README.md
Comment (ProfFan) [inline README.md:15]: Just wonder if this is still true, cause I am using 12.1 right now and it works fine?

File: README.md
Comment (ProfFan) [inline README.md:27]: Maybe we don't need this anymore. Let's delete all generation stuff cause this will cause confusion.

**Review by ProfFan — APPROVED** [review]
Great, thanks Dan! If you don't have time for the changes proposed I'll make a PR afterwards, just merge it :)

File: README.md
Comment (dan-zheng) [reply README.md]: Thanks for pointing this out! I updated the blurb about Xcode version to be more accurate:

> Download a [Swift for TensorFlow macOS toolchain](https://github.com/tensorflow/swift/blob/main/Installation.md#development-snapshots) and follow the [macOS installation instructions](https://github.com/tensorflow/swift/blob/master/Installation.md#macos). Installing the latest development snapshot is recommended. [Check here](https://github.com/apple/swift/blob/main/docs/HowToGuides/GettingStarted.md#installing-dependencies) for the "required" version of Xcode for using nightly toolchains. Sometimes toolchains may also work with older Xcode minor versions than the one listed.

File: README.md
Comment (dan-zheng) [reply README.md]: Thanks for the note! [`swift package generate-xcodeproj` is indeed becoming deprecated](https://forums.swift.org/t/rfc-deprecating-generate-xcodeproj/42159), though folks sometimes fall back on it instead of using Xcode's SwiftPM support.

I moved the "Xcode development" section under the `## Development` section alongside `### Visual Studio Code`.

**Comment by dan-zheng** [comment]
> Great, thanks Dan! If you don't have time for the changes proposed I'll make a PR afterwards, just merge it :)

Sure thing! Thanks for your comments - I believe I've addressed them, and I just realized I have merge access, so I'll merge when tests pass.

**Pull request merged**
